src/authentication/authentication-hapi-plugin.ts

```diff
@@ -202,14 +223,29 @@ class AuthenticationHapiPlugin extends HapiPlugin {
       const userInfo = await getAuth0UserInfo(this.hapiOptions.verifyOptions.issuer, accessToken, this.fetch);
       if (validateEmail(userInfo.email)) {
         email = userInfo.email;
-      // the email can actually be in the name field depending on the identity provider
+        // the email can actually be in the name field depending on the identity provider
       } else if (validateEmail(userInfo.name)) {
         email = userInfo.name;
       }
     }
     return email;
   }
 
+  private async teamIdOrNameToTeamId(teamIdOrName: string) {
+    let teamId = parseInt(String(teamIdOrName), 10);
+    if (isNaN(teamId)) {
```

Comment:
It could be a number, it could be a string containing a number or it could be a string containing a team name. With that parseInt construction I can handle all three cases and be sure I get a number if it's possible.